### PR TITLE
fix: edit CLI help for brevity, improve error feedback

### PR DESCRIPTION
**Pull request by wkillerud**
Part of improving feedback to users of the CLI.

Closes https://github.com/eik-lib/issues/issues/9

### CODE REVIEW

File: commands/alias.js [hunk 1]
```diff
@@ -1,85 +1,66 @@
...
+			server: {
+				alias: "s",
+				describe: "Eik server address, if different from configuration file",
+			},
+			type: {
+				describe:
+					"Alter the alias type. Default is detected from eik.json. Valid values are `package`, `npm`, or `map` Eg. --type npm",
+			},
+			token: {
+				describe: "JTW used for authentication, if not using eik login",
```
Comment (digitalsadhu) [inline commands/alias.js:39]:
```suggestion
				describe: "JWT used for authentication, if not using eik login",
```

File: commands/alias.js [hunk 2]
```diff
@@ -1,85 +1,66 @@
...
-	yargs.example(`eik alias my-app 4.2.2 4 --type package`);
+		})
+		.options({
+			server: {
+				alias: "s",
+				describe: "Eik server address, if different from configuration file",
+			},
+			type: {
+				describe:
+					"Alter the alias type. Default is detected from eik.json. Valid values are `package`, `npm`, or `map` Eg. --type npm",
```
Comment (digitalsadhu) [inline commands/alias.js:36]:
```suggestion
					"Alter the alias type. Default is detected from eik.json. Valid values are `package`, `npm`, 'image', or `map` Eg. --type npm",
```

File: commands/map-alias.js [hunk 3]
```diff
@@ -1,72 +1,68 @@
...
+				"Alias value, the semver major component of the import map version",
+			type: "string",
+		})
+		.options({
+			server: {
+				alias: "s",
+				describe: "Eik server address, if different from configuration file",
+			},
+			token: {
+				describe: "JTW used for authentication, if not using eik login",
```
Comment (digitalsadhu) [inline commands/map-alias.js:36]:
```suggestion
				describe: "JWT used for authentication, if not using eik login",
```

File: commands/map.js [hunk 4]
```diff
@@ -27,41 +26,40 @@ export const builder = (yargs) => {
...
-		`eik map my-map 2.1.1 ./import-map.json --server https://assets.myeikserver.com`,
-	);
+		})
+		.options({
+			server: {
+				alias: "s",
+				describe: "Eik server address, if different from configuration file",
+			},
+			token: {
+				describe: "JTW used for authentication, if not using eik login",
```
Comment (digitalsadhu) [inline commands/map.js:36]:
```suggestion
				describe: "JWT used for authentication, if not using eik login",
```

File: commands/npm-alias.js [hunk 5]
```diff
@@ -1,20 +1,19 @@
...
-import { logger, getDefaults } from "../utils/index.js";
+import { logger, getArgsOrDefaults } from "../utils/index.js";
 import { Alias as AliasFormatter } from "../formatters/index.js";
 
 export const command = "npm-alias <name> <version> <alias>";
 
 export const aliases = ["na", "dep-alias", "dependency-alias"];
 
-export const describe = `DEPRECATED: This command has been replaced by the alias command and will be removed in a future version. Create a semver major alias for an NPM package as identified by its name and version. An NPM package with the given name and version must already exist on the asset server. Alias should be the semver major part of the NPM package version. Eg. For an NPM package of version 5.4.3, you should use 5 as the alias`;
+export const describe = "Create an alias for an npm package";
```
Comment (digitalsadhu) [inline commands/npm-alias.js:10]:
```suggestion
export const describe = "Create an alias for an NPM package";
```

File: commands/map-alias.js [hunk 6]
```diff
@@ -1,72 +1,68 @@
...
 export const command = "map-alias <name> <version> <alias>";
 
 export const aliases = ["ma"];
 
-export const describe = `DEPRECATED: This command has been replaced by the alias command and will be removed in a future version. Create a semver major alias for an import map as identified by its name and version. An import map with the given name and version must already exist on asset server. Alias should be the semver major part of the import map version. Eg. For an import map of version 5.4.3, you should use 5 as the alias`;
+export const describe = "Create an alias for a map";
 
-export const builder = (yargs) => {
-	const defaults = getDefaults(yargs.argv.config || yargs.argv.cwd);
+export const deprecated = "map-alias is replaced by alias";
```
Comment (digitalsadhu) [inline commands/map-alias.js:12]:
```suggestion
export const deprecated = "map-alias has been replaced by alias";
```

File: commands/npm-alias.js [hunk 7]
```diff
@@ -1,20 +1,19 @@
...
 
-export const builder = (yargs) => {
-	const defaults = getDefaults(yargs.argv.config || yargs.argv.cwd);
+export const deprecated = "npm-alias is replaced by alias";
```
Comment (digitalsadhu) [inline commands/npm-alias.js:12]:
```suggestion
export const deprecated = "npm-alias has been replaced by alias";
```

File: commands/npm-alias.js [hunk 8]
```diff
@@ -27,45 +26,39 @@ export const builder = (yargs) => {
...
-		`eik npm lit-html 5.3.2 5 --server https://assets.myeikserver.com`,
-	);
+		})
+		.options({
+			server: {
+				alias: "s",
+				describe: "Eik server address, if different from configuration file",
+			},
+			token: {
+				describe: "JTW used for authentication, if not using eik login",
```
Comment (digitalsadhu) [inline commands/npm-alias.js:36]:
```suggestion
				describe: "JWT used for authentication, if not using eik login",
```

File: commands/package-alias.js [hunk 9]
```diff
@@ -1,80 +1,67 @@
...
 export const command = "package-alias [name] [version] [alias]";
 
 export const aliases = ["pkg-alias", "pa"];
 
-export const describe = `DEPRECATED: This command has been replaced by the alias command and will be removed in a future version. Create a semver major alias for a package as identified by its name and version. A package with the given name and version must already exist on asset server. Alias should be the semver major part of the package version. Eg. For a package of version 5.4.3, you should use 5 as the alias`;
+export const describe = "Create an alias for a package";
 
-export const builder = (yargs) => {
-	const defaults = getDefaults(yargs.argv.config || yargs.argv.cwd);
+export const deprecated = "package-alias is replaced by alias";
```
Comment (digitalsadhu) [inline commands/package-alias.js:12]:
```suggestion
export const deprecated = "package-alias has been replaced by alias";
```

File: commands/package-alias.js [hunk 10]
```diff
@@ -1,80 +1,67 @@
...
-	);
-	yargs.example(`eik package-alias my-app 4.2.2 4 --debug`);
+		})
+		.options({
+			server: {
+				alias: "s",
+				describe: "Eik server address, if different from configuration file",
+			},
+			token: {
+				describe: "JTW used for authentication, if not using eik login",
```
Comment (digitalsadhu) [inline commands/package-alias.js:36]:
```suggestion
				describe: "JWT used for authentication, if not using eik login",
```

File: commands/ping.js [hunk 11]
```diff
@@ -1,30 +1,28 @@
...
-		// @ts-expect-error
-		default: defaults.server,
-	});
-
-	yargs.example(`eik ping`);
-	yargs.example(`eik ping http://assets.myeikserver.com`);
-	yargs.example(`eik ping http://assets.myeikserver.com --debug`);
+	return yargs
+		.positional("server", {
+			describe: "Specify location of Eik server to ping.",
```
Comment (digitalsadhu) [inline commands/ping.js:16]:
```suggestion
			describe: "Specify location of Eik server to check against.",
```

File: commands/publish.js [hunk 12]
```diff
@@ -2,50 +2,56 @@ import { join } from "path";
...
-	yargs.example(`eik pkg --debug`);
+	return yargs
+		.options({
+			dryRun: {
+				alias: "d",
+				describe: "Log details about the operation and skip upload",
+				type: "boolean",
+			},
+			token: {
+				describe: "JTW used for authentication, if not using eik login",
```
Comment (digitalsadhu) [inline commands/publish.js:29]:
```suggestion
				describe: "JWT used for authentication, if not using eik login",
```

File: index.js [hunk 13]
```diff
@@ -1,84 +1,88 @@
...
+	.usage(
+		`usage: $0 [-v | --version] [-h | --help] [-c <path> | --config <path>]
+           [--cwd <path>] [--debug] <command> [<args>]`,
+	)
+	.epilogue(
+		`Run $0 <command> --help to read more about a specific subcommand.
+
+For a more detailed description, see the reference documentation:
+https://eik.dev/docs/reference/at-eik-cli/`,
+	)
```
Comment (digitalsadhu) [inline index.js:55]: oooh I like this

Comment (wkillerud) [reply index.js:55]: I had a brainwave this weekend to make it a bit smoother too. I'll add a redirect to this page from `eik.dev/cli` 🚀

**Comment by wkillerud** [comment]
```sh
node index.js alias
✖ No eik.json or package.json with eik configuration in /Users/william.killerud/workspace/eik/cli, and did not get required fields as options
```

**Pull request merged**